helm/dex-app/templates/dex-k8s-authenticator/configmap-customer.yaml

```diff
@@ -1,4 +1,58 @@
 {{ if .Values.oidc.customer.enabled }}
+{{- if .Values.oidc.customer.connectors -}}
+{{ $values := .Values }}
+{{ $resource := . }}
+{{- range .Values.oidc.customer.connectors }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "resource.dexk8sauth.name" $resource }}-{{ .id }}
+  labels:
+    {{- include "dexk8sauth.labels.common" $resource | nindent 4 }}
+data:
+  config.yaml: |-
+    listen: http://0.0.0.0:5555
+    web_path_prefix: "/"
+    debug: "false"
+    logo_uri: {{ $values.logoURI }}
+    {{- if not $values.ingress.tls.letsencrypt }}
+    trusted_root_ca_file: /app/ssl/ca.crt
+    {{- end }}
+    clusters:
+    {{- if $values.isManagementCluster }}
+    - name: {{ $values.managementCluster.name }}
+      description: "Management Cluster Kubernetes API for installation '{{ $values.managementCluster.name }}', as a Giant Swarm customer"
+      short_description: "{{ $values.managementCluster.name }} management cluster"
+      client_id: dex-k8s-authenticator
+      client_secret: {{ $values.oidc.staticClients.dexK8SAuthenticator.clientSecret }}
+      connector_id: {{ .id }}
+      issuer: https://{{ $values.oidc.issuerAddress }}
+      {{- if $values.services.kubernetes.api.public }}
+      k8s_master_uri: {{ $values.services.kubernetes.api.address }}
+      {{- else }}
+      k8s_master_uri: {{ $values.services.kubernetes.api.internalAddress }}
+      {{- end }}
+      k8s_ca_pem: {{ toYaml $values.services.kubernetes.api.caPem | indent 8 }}
+      redirect_uri: https://{{ $values.oidc.staticClients.dexK8SAuthenticator.clientAddress }}/callback
+    {{- else if $values.isWorkloadCluster }}
+    - name: {{ $values.clusterID }}
+      description: "Cluster Kubernetes API {{ $values.clusterID }}"
+      short_description: "{{ $values.clusterID }} cluster"
+      client_id: dex-k8s-authenticator
+      client_secret: {{ $values.oidc.staticClients.dexK8SAuthenticator.clientSecret }}
+      connector_id: {{ .id }}
+      issuer: https://dex.{{ $values.baseDomain }}
+      {{- if $values.services.kubernetes.api.public }}
+      k8s_master_uri: https://api.{{ $values.baseDomain }}
+      {{- else }}
+      k8s_master_uri: {{ $values.services.kubernetes.api.internalAddress }}
+      {{- end }}
+      k8s_ca_pem: {{ toYaml $values.services.kubernetes.api.caPem | indent 8 }}
+      redirect_uri: https://login.{{ $values.baseDomain }}/callback
+    {{- end }}
+{{- end }}
+{{- else }}
```

Comment:
It would be ideal, in terms of mgmt and clarity, to remove this second part of condition but I guess it will imply a migration for WC/MC deployments to use connectors array instead

Reply:
I agree with you here! :) We should slowly deprecate the old fields.
Moving our own config should not be such a big issue. However, we need to make sure not to break customers apps.